src/util/string_utils.cpp

```diff
@@ -158,7 +158,7 @@ std::string escape_non_alnum(const std::string &to_escape)
   {
     if(ch == '_')
       escaped << "__";
-    else if(isalnum(ch))
+    else if(ch >= 0 && isalnum(ch))
```

Comment:
I'm a little worried about this because certainly in C whether `char` is signed or not is implementation dependent.

Reply:
Now done through a `narrow_cast` cast, in order to avoid explicitly checking for negative values.